src/DatePicker/DatePicker.js

```diff
@@ -206,13 +209,55 @@ class DatePicker extends Component {
         date: date,
       });
     }
+
+    setTimeout(() => {
+      this.setState({
+        isFocused: false,
+      });
+    }, 0);
+    this.refs.input.focus();
+
     if (this.props.onChange) {
       this.props.onChange(null, date);
     }
   };
 
+  handleBlur = (event, shouldSetFocus) => {
+    if (shouldSetFocus) {
+      this.refs.input.focus();
+    }
+
+    if (this.state.isFocused) {
+      this.setState({
+        isFocused: false,
+      }, this.refs.dialogWindow.dismiss);
+    }
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  };
+
+  handleDismiss = () => {
+    if (this.state.isFocused) {
+      this.refs.input.focus();
+      this.setState({
+        isFocused: false,
+      });
+    }
+
+    if (this.props.onDismiss) {
+      this.props.onDismiss();
+    }
+  };
+
   handleFocus = (event) => {
-    event.target.blur();
+    if (!this.props.disabled && !this.state.isFocused) {
```

Comment:
For it to be removed the `onTouchTap` handler on the `TextField` would need to be removed, which alters the documentation. As the `onFocus` also takes care of `onTouchTap` I think you're right and it should be removed, but requires a change away from the documentation. How does handle things like this? 

Should the `onTouchTap` listener be removed from the `TextField`, but still handle the `onTouchTap` from the `props`? This could then be removed after a certain period of time, dependent upon the project's feature-deprecation time span. The problem with this is that the event being passed to the user's listener will not be the event type that they are expecting.